gears/src/baseapp/params.rs

```diff
@@ -293,4 +302,91 @@ mod tests {
                 .to_string()
         );
     }
+
+    #[derive(strum::EnumIter, Debug, PartialEq, Eq, Hash, Clone)]
+    enum SubspaceKey {
+        Params,
+    }
+
+    impl FromStr for SubspaceKey {
+        type Err = SubspaceParseError;
+
+        fn from_str(_: &str) -> Result<Self, Self::Err> {
+            Err(SubspaceParseError("empty enum".to_string()))
+        }
+    }
+
+    impl ParamsSubspaceKey for SubspaceKey {
+        fn name(&self) -> &'static str {
+            match self {
+                SubspaceKey::Params => "params",
+            }
+        }
+    }
+
+    impl StoreKey for SubspaceKey {
+        fn name(&self) -> &'static str {
+            "baseapp"
+        }
+
+        fn params() -> &'static Self {
+            const PARAM_KEY: SubspaceKey = SubspaceKey::Params;
+
+            &PARAM_KEY
+        }
+    }
+
+    #[test]
+    fn app_hash() {
+        let keeper = BaseAppParamsKeeper {
+            params_subspace_key: SubspaceKey::Params,
+        };
+
+        let mut multi_store = ApplicationMultiBank::<_, SubspaceKey>::new(MemDB::new());
+
+        let before_hash = multi_store.head_commit_hash();
+
+        let mut ctx = InitContext::new(
+            &mut multi_store,
+            0,
+            tendermint::types::time::timestamp::Timestamp::UNIX_EPOCH,
+            tendermint::types::chain_id::ChainId::default(),
+        );
+
+        keeper.set_consensus_params(&mut ctx, ConsensusParams::default());
+
+        multi_store.commit();
+        let after_hash = multi_store.head_commit_hash();
+
+        assert_ne!(before_hash, after_hash);
+
+        let expected_hash = [
+            139, 30, 111, 121, 185, 80, 199, 158, 15, 181, 206, 115, 179, 223, 81, 183, 11, 85, 80,
+            14, 41, 195, 81, 139, 165, 139, 13, 128, 138, 187, 254, 129,
+        ];
+
+        assert_eq!(expected_hash, after_hash);
+    }
+
+    #[test]
+    fn set_read_works() {
+        let keeper = BaseAppParamsKeeper {
+            params_subspace_key: SubspaceKey::Params,
+        };
+
+        let mut multi_store = ApplicationMultiBank::<_, SubspaceKey>::new(MemDB::new());
+
+        let mut ctx = InitContext::new(
```

Comment:
It seems like you may reuse your method in `utils`. I might be wrong because I don't know about configuration features here.

Reply:
This isn't possible. `utils` under feature while this compiles always